query/sources/common/src/main/java/org/codice/ddf/admin/sources/utils/RequestUtils.java

```diff
@@ -122,23 +121,22 @@
      * @param webClient {@code WebClient} to send POST request with
      * @param urlField  original request url field
      * @param content   Body of the post request
-     * @return a {@link ReportWithResultImpl} containing a {@link ResponseField} or an {@link org.codice.ddf.admin.api.report.ErrorMessage} on failure.
+     * @return a {@link Report} containing a {@link ResponseField} or an {@link org.codice.ddf.admin.api.report.ErrorMessage} on failure.
      */
-    public ReportWithResultImpl<ResponseField> sendPostRequest(WebClient webClient,
+    public Report<ResponseField> sendPostRequest(WebClient webClient,
             UrlField urlField, String content) {
-        ReportWithResultImpl<ResponseField> responseResult = new ReportWithResultImpl<>();
-        responseResult.addMessages(endpointIsReachable(urlField));
-        if (responseResult.containsErrorMsgs()) {
+        Report<ResponseField> responseResult = Reports.fromErrors(endpointIsReachable(urlField));
+        if (responseResult.containsErrorMessages()) {
             return responseResult;
         }
 
         try {
             Response response = webClient.post(content);
-            responseResult.result(responseFieldFromResponse(response, urlField));
+            responseResult.setResult(responseFieldFromResponse(response, urlField));
             return responseResult;
 
         } catch (ProcessingException e) {
-            responseResult.addArgumentMessage(cannotConnectError(urlField.path()));
+            responseResult.addErrorMessage(cannotConnectError(urlField.path()));
```

Comment:
Line 139-140 should be able to be `return Reports.from(responseFieldFromResponse(response, urlField));`